Extend dependency status box to report more issues

Enet4 opened the pull request:
I noticed that the dependency issue box presented at a project's status page may sometimes not present the most important information. In the example below, the project has 3 non-dev dependencies outdated, yet this box only reported the outdated dev dependencies.

![Screenshot_2021-10-12_11-06-06](https://user-images.githubusercontent.com/4738426/136984427-417e3d21-8e4c-4878-8241-9a77dd3a6493.png)

As such, I propose with this PR to extend this box to also provide information about outdated main dependencies, in a way which does not occlude the other kinds of issues. I also fixed pluralization, so that a single bad dependency appears in its singular form.

![Screenshot_2021-10-12_16-04-35](https://user-images.githubusercontent.com/4738426/136984524-4f87b40c-0b07-48c7-8664-a222095ef4f5.png)

Summary:

- replace render_dev_dependency_box with an extended render_dependency_box
   - reports insecure dev dependencies, outdated main dependencies, and outdated dev dependencies
   - handle pluralization in dependency count message
- change methods in `AnalyzeDependenciesOutcome`
   - add `count_outdated`
   - remove `any_dev_issues`, no longer used
- remove `AnalyzedDependencies::any_dev_issues`, no longer used

cecton commented:
@Enet4 I hope you don't mind, I pushed a commit with what I had in mind. I think it's more readable (on the page I mean) and the code is more concise. Let me know your opinion but feel free to revert if you don't like!! 

![image](https://user-images.githubusercontent.com/3018448/137468760-a5bb48ef-f7e5-435b-8457-1255cfdd461f.png)

![image](https://user-images.githubusercontent.com/3018448/137468801-423f1bd0-54e7-4a02-b360-406507aaae07.png)

Enet4 commented:
> I can already see an innocent dev removing that verification and no error will be thrown XD

That is fair, I can put this on hold until a safeguard is added.

Enet4 commented:
Great news (sort of)! This PR had one serious bug related with the counting of outdated main dependencies and it's now fixed here. I also addressed the wide assumption of `render_dependency_box` so that it is now called whenever no security issues were found.
